mungegithub/mungers/sig-mention-handler_test.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"reflect"
+	"runtime"
+	"testing"
+
+	githubapi "github.com/google/go-github/github"
+	github "k8s.io/test-infra/mungegithub/github"
+)
+
+const (
+	commentMessage = "There are no @kubernetes/sig mention on the issue. Please mention one."
+	helpWanted     = "help-wanted"
+	open           = "open"
+	otherVersion   = "v1.6"
+	sigApps        = "sig/apps"
+)
+
+func TestSigMentionHandler(t *testing.T) {
+	runtime.GOMAXPROCS(runtime.NumCPU())
+
+	tests := []struct {
+		name     string
+		obj      *github.MungeObject
+		expected bool
+	}{
+		{
+			name: "is a pull request",
+			obj: &github.MungeObject{
+				Issue: &githubapi.Issue{
+					State:     githubapi.String(open),
+					Milestone: &githubapi.Milestone{Title: githubapi.String(currentVersion)},
+					Labels: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+						githubapi.Label{Name: githubapi.String(sigApps)}},
+					PullRequestLinks: &githubapi.PullRequestLinks{},
+				},
+			},
+			expected: false,
+		},
+		{
+			name:     "issue is null",
+			obj:      &github.MungeObject{},
+			expected: false,
+		},
+		{
+			name: "issue state is null",
+			obj: &github.MungeObject{
+				Issue: &githubapi.Issue{
+					State:     nil,
+					Milestone: &githubapi.Milestone{Title: githubapi.String(currentVersion)},
+					Labels: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+						githubapi.Label{Name: githubapi.String(sigApps)}},
+					PullRequestLinks: nil,
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "issue is closed",
+			obj: &github.MungeObject{
+				Issue: &githubapi.Issue{
+					State:     githubapi.String(closed),
+					Milestone: &githubapi.Milestone{Title: githubapi.String(currentVersion)},
+					Labels: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
+						githubapi.Label{Name: githubapi.String(sigApps)}},
+					PullRequestLinks: nil,
+				},
+			},
+			expected: false,
+		},
+		{
+			name: "milestone is not v1.7",
```

Comment:
I'm pretty sure your code catches this case, but add a test case for no milestone set (many of our issues do not have an associated milestone)